test/uk/gov/hmrc/economiccrimelevyregistration/forms/behaviours/DateBehaviours.scala

```diff
@@ -41,6 +47,32 @@ class DateBehaviours extends FieldBehaviours {
           result.errors         shouldBe empty
         }
       }
+
+      "fail to bind invalid data" in {
+
+        val data = Map(
+          s"$key.day"   -> "dd",
+          s"$key.month" -> "MM",
+          s"$key.year"  -> "yyyy"
+        )
+
+        val result = form.bind(data)
+
+        result.errors should contain only FormError(key, invalidKey, errorArgs)
+      }
+
+      "fail to bind data out-of-range" in {
```

Comment:
I think I understand this test - but I think the description is a little misleading, it's not actually testing anything to do with the date range (min/max date). It's testing that if you provide an invalid day, month, or year then it shouldn't bind even if they are valid 'integers'.

the test is very similar to the one above which is testing that it shouldn't bind if the day/month/year is not even an integer, i think you could combine them if you defined an `arbInvalidLocalDate` which generated 3 strings, the day, month and year

Reply:
👍